reader_index.go

```diff
@@ -76,11 +77,13 @@ func (f *rsfReader) ReadIndex(r io.Reader) (Index, error) {
 		sz = int(binary.LittleEndian.Uint32(size))
 	}
 
-	f.index, err = f.readIndexEntries(r, sz, 0)
+	// Position when done reading index will be the current reader position +
+	// the index size, minus the size field length, since we've already read it.
+	f.index, err = f.readIndexEntries(r, f.pos+sz-sizeFieldLen, 0)
```

Comment:
The second part of the bug was that we passed the index size to the `readIndexEntries` method. This worked fine for V1 manifests, but for V2 manifests, the index size doesn't correctly indicate the ending reader position after reading the index.

For `V1` manifests, `f.pos` and `sizeFieldLen` will be equal at this point, so we're still essentially passing `sz` to the method. But for `V2` manifests, we're passing a different value that correctly indicates the ending position for reading the index.

The remaining logic (below) didn't change, but I renamed the `sz` variable for better clarity.